sdk/attestation/attestation/src/attestationAdministrationClient.ts

```diff
@@ -276,20 +321,278 @@ export class AttestationAdministrationClient {
+        addCertToken.serialize(),
+        updatedOptions
+      );
+      // The attestation token returned from the service has a PolicyResult
+      // object as the body.
+      const token = new AttestationTokenImpl(addCertificateResult.token);
+      token.validateToken(
+        await this.signingKeys(),
+        options.validationOptions ?? this._validationOptions
+      );
+
+      // Deserialize the PolicyCertificatesModificationResult object.
+      const result = TypeDeserializer.deserialize(
+        token.getBody(),
+        {
+          PolicyCertificatesModificationResult: Mappers.PolicyCertificatesModificationResult,
+          JsonWebKeySet: Mappers.JsonWebKeySet,
+          JsonWebKey: Mappers.JsonWebKey
+        },
+        "PolicyCertificatesModificationResult"
+      ) as PolicyCertificatesModificationResult;
+
+      return createAttestationResponse<PolicyCertificatesModificationResult>(token, result);
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  private keyTypeFromCertificate(cert: any): string {
+    let kty: string;
+    switch (cert.getSignatureAlgorithmName()) {
+      case "SHA256withRSA":
+      case "SHA384withRSA":
+      case "SHA512withRSA":
+        kty = "RSA";
+        break;
+      case "SHA256withECDSA":
+      case "SHA384withECDSA":
+        kty = "EC";
+        break;
+      default:
+        kty = "RSA";
+        break;
+    }
+    return kty;
+  }
+
+  /** Add a new certificate chain to the set of policy management certificates.
+   *
+   * @param pemCertificate - PEM encoded certificate to add to the set of policy management certificates.
+   * @param privateKey - Existing attestation private key used to sign the incoming request.
+   * @param certificate - Existing attestation certificate used to verify the incoming request.
+   * @param options - Options used in the call to the service.
+   * @returns An attestation response including a PolicyCertificatesModificationResult
+   *
+   * @remarks This API is only supported on `isolated` attestation instances.
+   *
+   * The signing key MUST be one of the existing attestation signing certificates. The
+   * new pemCertificate is signed using the signingKey and the service will validate the
+   * signature before allowing the addition.
+   *
+   */
+  public async removePolicyManagementCertificate(
+    pemCertificate: string,
+    privateKey: string,
+    certificate: string,
+    options: AttestationAdministrationClientOperationOptions = {}
+  ): Promise<AttestationResponse<PolicyCertificatesModificationResult>> {
+    const { span, updatedOptions } = createSpan(
+      "AttestationAdministrationClient-removePolicyManagementCertificate",
+      options
+    );
+    try {
+      if ((!privateKey && certificate) || (privateKey && !certificate)) {
+        throw new TypeError(
```

Comment:
@xirzec I do not remember from the guidelines if it is ok to throw `TypeError`s. Have you seen this before?

Reply:
`TypeError` seems to be the error thrown when doing client side validation, but I'm completely open to using a different error (including `Error`, if necessary).